data/serializer/src/main/java/tech/pegasys/teku/api/ssz/DepositTreeSnapshotSerializer.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright ConsenSys Software Inc., 2022
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.api.ssz;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.apache.tuweni.ssz.SSZ;
+import tech.pegasys.teku.api.data.DepositTreeSnapshotData;
+import tech.pegasys.teku.infrastructure.http.RestApiConstants;
+import tech.pegasys.teku.infrastructure.restapi.openapi.response.OctetStreamResponseContentTypeDefinition;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+
+public class DepositTreeSnapshotSerializer implements SszSerializer<DepositTreeSnapshotData> {
```

Comment:
hmm, I was originally expecting this to just be a serialiser for the database format (like we have `BlockCheckpointsSerializer`) but we're having to serialise to SSZ in the REST API and presumably at some point need to deserialise SSZ when the user specifies a snapshot on the command line.  Plus from what I can see the `DepositTreeSnapshot` is only ever used as a data type for this snapshot and serialised as either json or ssz.

So I think I've given you some bad advice and we should just make `DepositTreeSnapshot` an SSZ type and then we get the SSZ serialisation/deserialization plus the JsonTypeDefinition for "free" from that. Sorry.